pageserver/src/tenant/timeline/detach_ancestor.rs

```diff
@@ -116,6 +125,404 @@ impl Default for Options {
     }
 }
 
+/// SharedState manages the pausing of background tasks (GC) for the duration of timeline detach
+/// ancestor.
+///
+/// Currently this is tracked at tenant level, but it could be moved to be on the roots
+/// of each timeline tree.
+#[derive(Default)]
+pub(crate) struct SharedState {
+    inner: std::sync::Mutex<SharedStateInner>,
+    gc_waiting: tokio::sync::Notify,
+    attempt_waiting: tokio::sync::Notify,
+}
+
+impl SharedState {
+    /// Notify an uninitialized shared state that an attempt to detach timeline ancestor continues
+    /// from previous instance.
+    pub(crate) fn notify(&self, attempt: &Attempt) {
```

Comment:
notify is not a great name; something something about the existing attempt. `[continue_]existing_attempt`?